app/screens/gallery/footer/download_with_action/index.tsx

```diff
@@ -206,6 +206,11 @@ const DownloadWithAction = ({action, enableSecureFilePreview, item, onDownloadSu
     const saveImageOrVideo = async (path: string) => {
         if (mounted.current) {
             try {
+                const mimeType = lookupMimeType(item.name || path);
+
+                if (mimeType === 'application/octet-stream') {
+                    saveFile(path);
```

Comment:
```suggestion
                    saveFile(path);
                    return;
```

But this doesn't look to be in the right place